src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2StreamTests.cs

```diff
@@ -3103,15 +3103,11 @@ public async Task ResponseWithHeadersTooLarge_AbortsConnection()
         await InitializeConnectionAsync(async context =>
         {
             context.Response.Headers["too_long"] = new string('a', (int)Http2PeerSettings.DefaultMaxFrameSize);
-            var ex = await Assert.ThrowsAsync<InvalidOperationException>(() => context.Response.WriteAsync("Hello World")).DefaultTimeout();
-            appFinished.TrySetResult(ex.InnerException.Message);
+            await context.Response.WriteAsync("Hello World");
```

Comment:
This is a small behavior change. Is the new behavior intentional?

Reply:
Yes. There's no way to get a synchronous exception when writes are queued like this. I spoke to @Tratcher about it and he agreed this is fine.

Reply:
We could preemptively check if any outgoing header values are longer than the current value of `Http2FrameWriter._maxFrameSize`. It makes this error a little clearer to the developer if it gets thrown directly to their app vs just showing up in the logs.

I don't think it's worth it though because we'd have to iterate over the response headers and additional time every request just for this validation.